core: uncrustify

kaspar030 opened the pull request:
### Contribution description

This PR applies the current uncrustify-riot.cfg to core.
```atomic_*.c``` are excluded due to too many misformattings in the macros.

Most of the changes are alright. My main pain point are string concatenations for inttype format macros, e.g.,

```DEBUG("foo %"PRIu32"\n");```

changes into

```DEBUG("foo %" PRIu32 "\n");```

(note the extra spaces between ```"``` and ```PRIu32```.

But I can live with that.

### Testing procedure

Check diff.

### Issues/PRs references

- #8519.

- ~~needs #10873~~

miri64 commented:
> Most of the changes are alright. My main pain point are string concatenations for inttype format macros, e.g.,
> 
> `DEBUG("foo %"PRIu32"\n");`
> 
> changes into
> 
> `DEBUG("foo %" PRIu32 "\n");`
> 
> (note the extra spaces between `"` and `PRIu32`.
> 
> But I can live with that.

I actually find the auto-formatted version more readable, but again, a matter of taste.

kaspar030 commented:
> I actually find the auto-formatted version more readable, but again, a matter of taste.

Yup. I'm OK with the auto-formatted version, too.

kaspar030 commented:
I realized I'd forgotten the includes... They're uncrustified now, too.

kaspar030 commented on core/atomic_c11.c:
```diff
@@ -41,6 +41,11 @@
 
+/*
+ * uncrustify mis-formats the macros in this file, so disable it globally.
+ * *INDENT-OFF*
```
good idea, done.

miri64 commented on core/atomic_c11.c:
```diff
@@ -330,7 +337,7 @@ void __atomic_exchange_c(size_t size, void *ptr, void *val, void *ret, int memor
  */
 bool __atomic_compare_exchange_c(size_t len, void *ptr, void *expected,
-    void *desired, bool weak, int success_memorder, int failure_memorder)
+                                 void *desired, bool weak, int success_memorder, int failure_memorder)
```
This was delivered by `uncrustify` this way? `int failure_memorder` could (and should) be on the next line.

kaspar030 replied:
It seems like our uncrustify config doesn't do line wrapping at all. I've pushed a change to make it do it.

It feels quite intrusive... ;)

miri64 replied:
> It feels quite intrusive... ;)

If it formats it this way it's plain wrong though.

kaspar030 replied:
> If it formats it this way it's plain wrong though.

Hmyeah. I tried limiting uncrustify's line wrap to 100 cols. It prevents this mishap.

kaspar030 replied:
An alternative is setting ```ls_func_split_full```, which results in this (if a line is longer than the configured limit):

```
-bool __atomic_compare_exchange_c(size_t len, void *ptr, void *expected,
-                                 void *desired, bool weak, int success_memorder, int failure_memorder)
+bool __atomic_compare_exchange_c(size_t len,
+                                 void *ptr,
+                                 void *expected,
+                                 void *desired,
+                                 bool weak,
+                                 int success_memorder,
+                                 int failure_memorder)
```

miri64 replied:
Don't like that style (and only use it if everything else doesn't make sense stylistically)

kaspar030 replied:
I've split out the 100char config change into another PR: #10873

kaspar030 commented:
- rebased
- cleaned first commit (contained both disabling of uncrustify and some formatting changes
- re-run uncrustify with #10873 (80char line limit) merged

@miri64 should we go forward with this?

miri64 commented:
I do think so. I put it on my TODO list ;-).

kaspar030 commented:
*ping* this is purely cosmetic. :)

miri64 commented on core/atomic_sync.c:
```diff
@@ -29,6 +29,11 @@
 
+/*
+ * uncrustify mis-formats the macros in this file, so disable it globally.
+ * begin{code-style-ignore}
```
`/* end{code-style-ignore} */` missing?

kaspar030 replied:
Well, I could add it at the end. The whole file is supposed to be ignored.

miri64 replied:
Even though it might not be checked, I think its cleaner to always have an end to a beginning ;-).

miri64 commented on core/bitarithm.c:
```diff
@@ -31,10 +31,10 @@ unsigned bitarithm_msb(unsigned v)
+    shift = (v > 0xFF) << 3; v >>= shift; r |= shift;
+    shift = (v > 0xF) << 2; v >>= shift; r |= shift;
+    shift = (v > 0x3) << 1; v >>= shift; r |= shift;
+    r |= (v >> 1);
```
I remember that some people wanted the original formatting (I also think it looks quite nice, but it of course breaks any automation). Maybe surround it with `/* begin{code-style-ignore} */`/`/* end{code-style-ignore} */`?

miri64 commented on core/include/kernel_defines.h:
```diff
@@ -42,21 +42,21 @@
+              ((TYPE *)((char *)(PTR)-offsetof(TYPE, MEMBER))), \
+              __typeof__ (((TYPE *)0)->MEMBER) * : \
+              ((TYPE *)((char *)(PTR)-offsetof(TYPE, MEMBER))) \
+              ))
```
This is somewhat hard to read, but I'm pretty sure the formatter broke some formatting here.

miri64 commented on core/include/kernel_defines.h:
```diff
@@ -42,21 +42,21 @@
+    (__extension__({ \
+        __extension__ const __typeof__ (((TYPE *)0)->MEMBER) * __m____ = (PTR); \
+        ((TYPE *)((char *)__m____ - offsetof(TYPE, MEMBER))); \
+    }))
```
Here as well.

miri64 commented on core/include/kernel_defines.h:
```diff
@@ -42,21 +42,21 @@
 #else
 #   define container_of(PTR, TYPE, MEMBER) \
-        ((TYPE *) ((char *) (PTR) - offsetof(TYPE, MEMBER)))
+    ((TYPE *)((char *)(PTR)-offsetof(TYPE, MEMBER)))
```
This definitely now is broken (`-` should have spaces around it)

miri64 commented on core/include/mbox.h:
```diff
@@ -31,7 +31,8 @@ extern "C" {
 /** Static initializer for mbox objects */
-#define MBOX_INIT(queue, queue_size) {{0}, {0}, CIB_INIT(queue_size), queue}
+#define MBOX_INIT(queue, queue_size) { { 0 }, { 0 }, CIB_INIT(queue_size), \
+        queue }
```
I think I prefer

```C
#define MBOX_INIT(queue, queue_size) { \
        { 0 }, { 0 }, CIB_INIT(queue_size), queue \
    }
```

miri64 commented on core/include/sched.h:
```diff
@@ -122,7 +122,7 @@ typedef enum {
  */
 #define STATUS_ON_RUNQUEUE      STATUS_RUNNING  /**< to check if on run queue:
-                                                 `st >= STATUS_ON_RUNQUEUE`   */
+                                                   `st >= STATUS_ON_RUNQUEUE`   */
```
This seems off...

```suggestion
                                                     `st >= STATUS_ON_RUNQUEUE`   */
```

miri64 commented on core/panic.c:
```diff
@@ -41,13 +41,14 @@ const char assert_crash_message[] = "FAILED ASSERTION.";
 
-void __attribute__((weak)) panic_arch(void) {}
+void __attribute__((weak)) panic_arch(void)
+{}
```
:thinking: not sure this is more correct...

miri64 replied:
Proposal:

```C
void __attribute__((weak)) panic_arch(void)
{
    return;
}
```

More verbose, but less weird ;-)

miri64 commented on core/sched.c:
```diff
@@ -89,11 +91,14 @@ int __attribute__((used)) sched_run(void)
+        "sched_run: active thread: %" PRIkernel_pid ", next thread: %" PRIkernel_pid "\n",
+        (kernel_pid_t)((active_thread ==
+                        NULL) ? KERNEL_PID_UNDEF : active_thread->pid),
+        next_thread->pid);
```
This looks pretty messed up. How about:

```C
    DEBUG(
        "sched_run: active thread: %" PRIkernel_pid ", next thread: %" PRIkernel_pid "\n",
        (kernel_pid_t)((active_thread == NULL)
                      ? KERNEL_PID_UNDEF
                      : active_thread->pid),
        next_thread->pid);
```

This should also be accepted by `uncrustify`

miri64 replied:
Also (but unrelated to this PR): Is the cast really necessary?

miri64 commented:
> I'm about to add the ignore statements to the whole file. What do you think?

If it is properly ended, go ahead :-).

miri64 commented:
> There are some alignments in `core/include/byteorder.h` that seem broken. I'm about to add the ignore statements to the whole file. What do you think?

Mhh.. after looking at the current proposal, I'm not sure what you mean. Can you point me to that?

kaspar030 commented on core/include/byteorder.h:
```diff
@@ -48,9 +48,9 @@ typedef union __attribute__((packed)) {
-    uint16_t    u16[2]; /**< 16 bit representation */
+    uint32_t u32;       /**< 32 bit representation */
+    uint8_t u8[4];      /**< 8 bit representation */
+    uint16_t u16[2];    /**< 16 bit representation */
```
@miri64 here's an example of previously aligned changes.

miri64 replied:
Mh, but is the alignment really critical here? This looks more like some ASCII-art quirk of the original author. Compared to e.g. https://github.com/RIOT-OS/RIOT/pull/10867/files#diff-f4f930b62cff9df936337e8842e6c5d6R34 this doesn't really improve readability doesn't makes clearer what belongs where.

kaspar030 replied:
ok, then let's uncrustify be our new ASCII artist.

kaspar030 commented:
@miri64 I've rebased this, it is otherwise unchanged. Let's find a shorter you-look-I-react cycle.

miri64 commented on core/include/sched.h:
```diff
@@ -122,7 +122,8 @@ typedef enum {
 #define STATUS_ON_RUNQUEUE      STATUS_RUNNING  /**< to check if on run queue:
-                                                 `st >= STATUS_ON_RUNQUEUE`   */
+                                                     `st >= STATUS_ON_RUNQUEUE`
+                                                 */
```
This looks weird. Any idea why uncrustify is doing this?

kaspar030 replied:
I think actually I did this, must be a merge leftover. Will fix.

kaspar030 replied:
done. Now uncrustify just realignes the second line a bit.

kaspar030 commented:
- fixed comment, rebased

miri64 commented:
ACK! Let's finally merge this!

kaspar030 commented:
Thanks @miri64!
